### ISSUE TITLE
Change proposal: prefer-numeric-literals should also disallow Number.parseInt

### ISSUE THREAD

**Issue by kirly-af** [eslint/eslint#8913]
**What rule do you want to change?**
[prefer-numeric-literals](http://eslint.org/docs/rules/prefer-numeric-literals)

**Does this change cause the rule to produce more or fewer warnings?**
More.

**How will the change be implemented? (New option, new default behavior, etc.)?**
New default behavior.

**Please provide some example code that this change will affect:**

```js
/*eslint prefer-numeric-literals: "error"*/

Number.parseInt("111110111", 2) === 503;
Number.parseInt("767", 8) === 503;
Number.parseInt("1F7", 16) === 255;
```

**What does the rule currently do for this code?**
Nothing.

**What will the rule do after it's changed?**
Same behavior as the way `parseInt()` is handled:
Disallow `Number.parseInt()` if it is called with two arguments: a string and a radix option of 2 (binary), 8 (octal), or 16 (hexadecimal).

**Comment by platinumazure** [comment]
Seems reasonable, especially if we already have special logic for global `parseInt`. I'll champion this enhancement.

@eslint/eslint-team What do you think?

**Comment by platinumazure** [comment]
Working on this.
